libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1318,11 +1318,39 @@ bool TypeChecker::visit(Conditional const& _conditional)
 	return false;
 }
 
+void TypeChecker::checkExpressionAssignment(Type const& _type, Expression const& _expression)
+{
+	if (auto const* tupleExpression = dynamic_cast<TupleExpression const*>(&_expression))
+	{
+		if (auto const* tupleType = dynamic_cast<TupleType const*>(&_type))
+		{
+			for (size_t i = 0; i < min(tupleExpression->components().size(), tupleType->components().size()); i++)
+				if (tupleType->components()[i] && tupleExpression->components()[i])
+					checkExpressionAssignment(*tupleType->components()[i], *tupleExpression->components()[i]);
+		}
+		else if (!tupleExpression->components().empty())
```

Comment:
Would it perhaps be better to reduce this case to the other by creating a vector of types and thus only have one single place where we have a recursive call?